sys/cddl/dev/dtrace/dtrace_ioctl.c

```diff
@@ -373,23 +378,25 @@ dtrace_ioctl(struct cdev *dev, u_long cmd, caddr_t addr,
 		return (0);
 	}
 	case DTRACEIOC_DOFGET: {
-		dof_hdr_t **pdof = (dof_hdr_t **) addr;
-		dof_hdr_t hdr, *dof = *pdof;
+		//TODO(nicomazz): fix when userspace is not purecap
+		dof_hdr_t *__capability*pdof = (dof_hdr_t *__capability*) addr;
+		dof_hdr_t hdr, *__capability dof = *pdof;
 		int rval;
 		uint64_t len;
 
 		DTRACE_IOCTL_PRINTF("%s(%d): DTRACEIOC_DOFGET\n",__func__,__LINE__);
 
-		if (copyin((void *)dof, &hdr, sizeof (hdr)) != 0)
+		if (copyin_c(dof, &hdr, sizeof (hdr)) != 0)
 			return (EFAULT);
 
 		mutex_enter(&dtrace_lock);
-		dof = dtrace_dof_create(state);
+		dof = (__cheri_tocap  dof_hdr_t * __capability) dtrace_dof_create(state);
```

Comment:
Might be easier to read by using a different non-capability local var for the new dof pointer.